glue support (#225)
ExpediaGroup/waggle-dance: README.md

| `primary-meta-store.mapped-tables`                      | No       | List of mappings from databases to tables to federate from the primary metastore, similar to `mapped-databases`. By default, all tables are available. See `mapped-tables` configuration below. |
| `primary-meta-store.hive-metastore-filter-hook`        | No       | Name of the class which implements the `MetaStoreFilterHook` interface from Hive. This allows a metastore filter hook to be applied to the corresponding Hive metastore calls. Can be configured with the `configuration-properties` specified in the `waggle-dance-server.yml` configuration. They will be added in the HiveConf object that is given to the constructor of the `MetaStoreFilterHook` implementation you provide. |
| `primary-meta-store.database-name-mapping`             | No       | BiDirectional Map of database names and mapped name, where key=`<database name as known in the primary metastore>` and value=`<name that should be shown to a client>`. See the [Database Name Mapping](#database-name-mapping) section.|

> | `primary-meta-store.glue-config`             | No       | Can be used instead of `remote-meta-store-uris` to federate to an AWS Glue Catalog ([AWS Glue](https://docs.aws.amazon.com/glue/index.html). See the [Federate to AWS Glue Catalog](#federate-to-aws-glue-catalog) section.|

Reviewer: Super picky, I know, but can you line up the `|` chars on this line with the other lines so that reading the raw file is a bit easier?

| `federated-meta-stores[n].database-name-mapping`        | No       | BiDirectional Map of database names and mapped names where key=`<database name as known in the federated metastore>` and value=`<name that should be shown to a client>`. See the [Database Name Mapping](#database-name-mapping) section.|
| `federated-meta-stores[n].writable-database-white-list` | No       | White-list of databases used to verify write access used in conjunction with `federated-meta-stores[n].access-control-type`. The list of databases should be listed without a `federated-meta-stores[n].database-prefix`. This property supports both full database names and (case-insensitive) [Java RegEx patterns](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html).|

> | `ederated-meta-stores[n].glue-config`             | No       | Can be used instead of `remote-meta-store-uris` to federate to an AWS Glue Catalog ([AWS Glue](https://docs.aws.amazon.com/glue/index.html). See the [Federate to AWS Glue Catalog](#federate-to-aws-glue-catalog) section.|

Reviewer: ```suggestion
| `federated-meta-stores[n].glue-config`             | No       | Can be used instead of `remote-meta-store-uris` to federate to an AWS Glue Catalog ([AWS Glue](https://docs.aws.amazon.com/glue/index.html). See the [Federate to AWS Glue Catalog](#federate-to-aws-glue-catalog) section.|
```

|:----|:----:|:----|
| `overwrite-config-on-shutdown`    | No         | Controls whether the federations configuration must be overwritten when the server is stopped. Settings this to `false` will cause any federations dynamically added at runtime to be lost when the server is stopped. This is also the case of databases created at runtime when `database-resolution` is set to `MANUAL`. Default is `true`. |

#### Federate to AWS Glue Catalog

Waggle Dance supports federation to AWS Glue Catalog. The federation only works as read-only. Write (Create/Alter/Drop) operations are not supported very well as the Glue APIS don't expose all Hive Metastore functions for instance lock/transactions and other functions are not supported so clients might get exceptions when using certain operations (this can depend on a client like Hive, Spark, etc...). Some research has been done to allow write operations and it is not impossible with a bit more work but out of scope at the moment.
The GlueConfig configuration should be used if federation to Glue is needed.

| Property        | Required | Description |
|:----|:----:|:----|

> | `glue-account-id` | Yes (if glueConfig used)   | The AWS account number.|

Reviewer: ```suggestion
| `glue-account-id` | Yes (if `glueConfig` used)   | The AWS account number.|
```

> | `glue-endpoint`   | Yes (if glueConfig used)   | The AWS glue endpoint. Is the same for all accounts key, is the region as Glue catalogs are per account and per region.|

Reviewer: ```suggestion
| `glue-endpoint`   | Yes (if `glueConfig` used)   | The AWS glue endpoint. Is the same for all accounts key, is the region as Glue catalogs are per account and per region.|
```

Reviewer: > Is the same for all accounts key, is the region as Glue catalogs are per account and per region.

I'm not sure what this means.  Can we clarify?

Author: brainfarth, will make that more clear

 Example,:

    glue-config:
    	glue-account-id: 1234566789012

>       glue-endpoint: glue.us-east-1.amazonaws.com

Reviewer: Indentation seems messed up.

> As with Hive federation the permission need to be setup to read underlying data, that's not part of Waggle Dance. If federating across AWS accounts the correct (cross account federation}[https://docs.aws.amazon.com/glue/latest/dg/cross-account-access.html] needs to be setup as well.       

Reviewer: ```suggestion
As with Hive federation, the IAM permissions need to be setup to read underlying data. IAM permissions are not setup by this code, but are usually setup by the Terraform code that deploys WaggleDance, such as (apiary-federation)[https://github.com/ExpediaGroup/apiary-federation].

If federating across AWS accounts, the correct (cross account federation permissions)[https://docs.aws.amazon.com/glue/latest/dg/cross-account-access.html] needs to be setup as well.       
```
